### ISSUE TITLE
Window Covering does not work with homekit node

### ISSUE THREAD

**Issue by Nazarbayevv** [deconz-community/node-red-contrib-deconz#196]
### Current Behavior

Deconz outpoot node doesn't respond to homekit's node requests, when I use Homekit type of commands:
![image](https://user-images.githubusercontent.com/50654913/149541632-221087dc-2a86-450f-b94b-555925fd97f4.png)

### Expected Behavior

It works, when I use windows cover mode and set lift field as expression with velue: (100 - msg.payload.TargetPosition). But it's not very user frendly.
![image](https://user-images.githubusercontent.com/50654913/149543508-ca65b0b4-e2c2-4c26-9175-94d9270d404b.png)

### Steps To Reproduce

![image](https://user-images.githubusercontent.com/50654913/149543886-3b1197e6-4b44-4013-be00-8cdfb47e7648.png)

### Example flow
```json
[{"id":"4b90078f.78cc28","type":"homekit-service","z":"f7380f55.3bcd1","isParent":true,"bridge":"c887541e.9bc998","parentService":"","name":"Занавеска","serviceName":"WindowCovering","topic":"","filter":false,"manufacturer":"NRCHKB","model":"1.3.5","serialNo":"Default Serial Number","firmwareRev":"1.3.5","hardwareRev":"1.3.5","softwareRev":"1.3.5","cameraConfigVideoProcessor":"ffmpeg","cameraConfigSource":"","cameraConfigStillImageSource":"","cameraConfigMaxStreams":2,"cameraConfigMaxWidth":1280,"cameraConfigMaxHeight":720,"cameraConfigMaxFPS":10,"cameraConfigMaxBitrate":300,"cameraConfigVideoCodec":"libx264","cameraConfigAudioCodec":"libfdk_aac","cameraConfigAudio":false,"cameraConfigPacketSize":1316,"cameraConfigVerticalFlip":false,"cameraConfigHorizontalFlip":false,"cameraConfigMapVideo":"0:0","cameraConfigMapAudio":"0:1","cameraConfigVideoFilter":"scale=1280:720","cameraConfigAdditionalCommandLine":"-tune zerolatency","cameraConfigDebug":false,"cameraConfigSnapshotOutput":"disabled","cameraConfigInterfaceName":"","characteristicProperties":"{}","waitForSetupMsg":false,"outputs":2,"x":370,"y":4400,"wires":[["3a2e11926a079c60"],[]]},{"id":"684ae8fe7ab2f05b","type":"deconz-input","z":"f7380f55.3bcd1","name":"","topic":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","outputs":1,"output_rules":[{"type":"homekit","format":"single","onstart":true,"onerror":true,"payload":["CurrentPosition","TargetPosition"]}],"x":170,"y":4400,"wires":[["4b90078f.78cc28"]]},{"id":"3a2e11926a079c60","type":"deconz-output","z":"f7380f55.3bcd1","name":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","commands":[{"arg":{"payload":{"type":"msg","value":"payload"},"transitiontime":{"type":"num","value":""},"retryonerror":{"type":"num","value":"0"},"aftererror":{"type":"continue","value":""}},"type":"homekit","domain":""}],"specific":{"delay":{"type":"num","value":"50"},"result":{"type":"at_end","value":""}},"x":590,"y":4400,"wires":[[]]},{"id":"c887541e.9bc998","type":"homekit-bridge","bridgeName":"Node-RED_Access","pinCode":"400-44-000","port":"","allowInsecureRequest":false,"manufacturer":"NRCHKB","model":"1.2.0","serialNo":"Default Serial Number","firmwareRev":"1.2.0","hardwareRev":"1.2.0","softwareRev":"1.2.0","customMdnsConfig":false,"mdnsMulticast":true,"mdnsInterface":"","mdnsPort":"","mdnsIp":"","mdnsTtl":"","mdnsLoopback":true,"mdnsReuseAddr":true,"allowMessagePassthrough":false},{"id":"da76db0b.163948","type":"deconz-server","name":"Deconz","config_version":1,"ip":"192.168.88.9","port":"80","ws_port":"443","secure":false,"polling":"15"}]
```
### Environment

- deCONZ plugin version: 2.3.0
- HomeKit plugin version (if used): 1.4.3
- Node-RED version: 2.1.4
- Node.js version:
- npm version:
- Platform/OS: 
- Browser:

**Comment by Zehir** [comment]
The fix is published in 2.3.1

**Comment by Zehir** [comment]
Could you send the deconz events logs related to this beavior ?

**Comment by Nazarbayevv** [comment]
Logs from node red debug or logs from deconz-gui? Or any other logs?

**Comment by Zehir** [comment]
> Logs from node red debug or logs from deconz-gui? Or any other logs?
> 

Just plug an debug log after the input node to see what you get from deconz and also after the output node to see what you sent to it

**Comment by Nazarbayevv** [comment]
![image](https://user-images.githubusercontent.com/50654913/151587359-b6371e2c-3862-4929-b26d-5395d18554b6.png)
1-st debug:
28.01.2022, 18:45:54node: 1
msg.payload : Object
object
CurrentPosition: 1
TargetPosition: 1

2-nd debug:
28.01.2022, 18:45:52node: 2
msg.payload : Object
object
TargetPosition: 30

3-rd debug:
28.01.2022, 18:45:52node: 3
msg.payload : array[1]
array[1]
0: object
result: object
lift: 100
request: object
lift: 100
meta: object
etag: "130ab19bffcc733b9821605fc9686f9e"
hascolor: false
lastannounced: "2022-01-07T16:03:30Z"
lastseen: "2022-01-07T16:03Z"
manufacturername: "_TZE200_xuzcvlku"
modelid: "TS0601"
name: "Занавеска"
state: object
swversion: null
type: "Window covering device"
uniqueid: "5c:02:72:ff:fe:e1:60:3f-01"
device_type: "lights"
device_id: 44
device_path: "lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"

**Comment by Zehir** [comment]
Could you send an export of your flow ?

**Comment by Zehir** [comment]
Could you try this :

Close the windows cover.
Clear all logs
Open the cover with the homekit app.
Wait 10 seconds after the cover stop moving.
Then send all logs you get from the nodes.

**Comment by Zehir** [comment]
For the next debug logs could you use the copy value button ?
![image](https://user-images.githubusercontent.com/845225/151660104-04b34c60-0673-4346-bc2a-f201f527418a.png)

**Comment by Nazarbayevv** [comment]
Hello.

1. 


> Could you send an export of your flow ?

[{"id":"276cd89c880e0727","type":"deconz-input","z":"62a84518.fc2e2c","name":"","topic":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","outputs":1,"output_rules":[{"type":"homekit","format":"single","onstart":true,"onerror":true,"payload":["CurrentPosition","TargetPosition"],"uniqueId":"ui-id-11"}],"x":150,"y":480,"wires":[["c5e9ee34a4573cd3","37b4a55ad6577a83"]]},{"id":"0745d4f5296d7bae","type":"deconz-output","z":"62a84518.fc2e2c","name":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","commands":[{"arg":{"payload":{"type":"msg","value":"payload"},"transitiontime":{"type":"num","value":""},"retryonerror":{"type":"num","value":"0"},"aftererror":{"type":"continue","value":""}},"type":"homekit","domain":"","uniqueId":"ui-id-10"}],"specific":{"delay":{"type":"num","value":"50"},"result":{"type":"at_end","value":""}},"x":610,"y":480,"wires":[["908997aa4d3bfc5b"]]},{"id":"c5e9ee34a4573cd3","type":"debug","z":"62a84518.fc2e2c","name":"1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":300,"y":400,"wires":[]},{"id":"59fb1951dfad846a","type":"debug","z":"62a84518.fc2e2c","name":"2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":540,"y":400,"wires":[]},{"id":"908997aa4d3bfc5b","type":"debug","z":"62a84518.fc2e2c","name":"3","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":790,"y":400,"wires":[]},{"id":"37b4a55ad6577a83","type":"homekit-service","z":"62a84518.fc2e2c","isParent":true,"hostType":"0","bridge":"c887541e.9bc998","accessoryId":"","parentService":"","name":"test","serviceName":"WindowCovering","topic":"","filter":false,"manufacturer":"NRCHKB","model":"1.4.3","serialNo":"Default Serial Number","firmwareRev":"1.4.3","hardwareRev":"1.4.3","softwareRev":"1.4.3","cameraConfigVideoProcessor":"ffmpeg","cameraConfigSource":"","cameraConfigStillImageSource":"","cameraConfigMaxStreams":2,"cameraConfigMaxWidth":1280,"cameraConfigMaxHeight":720,"cameraConfigMaxFPS":10,"cameraConfigMaxBitrate":300,"cameraConfigVideoCodec":"libx264","cameraConfigAudioCodec":"libfdk_aac","cameraConfigAudio":false,"cameraConfigPacketSize":1316,"cameraConfigVerticalFlip":false,"cameraConfigHorizontalFlip":false,"cameraConfigMapVideo":"0:0","cameraConfigMapAudio":"0:1","cameraConfigVideoFilter":"scale=1280:720","cameraConfigAdditionalCommandLine":"-tune zerolatency","cameraConfigDebug":false,"cameraConfigSnapshotOutput":"disabled","cameraConfigInterfaceName":"","characteristicProperties":"{}","waitForSetupMsg":false,"outputs":2,"x":370,"y":480,"wires":[["59fb1951dfad846a","0745d4f5296d7bae"],[]]},{"id":"da76db0b.163948","type":"deconz-server","name":"Deconz","config_version":1,"ip":"192.168.88.9","port":"80","ws_port":"443","secure":false,"polling":"15"},{"id":"c887541e.9bc998","type":"homekit-bridge","bridgeName":"Node-RED_Access","pinCode":"400-44-000","port":"","allowInsecureRequest":false,"manufacturer":"NRCHKB","model":"1.2.0","serialNo":"Default Serial Number","firmwareRev":"1.2.0","hardwareRev":"1.2.0","softwareRev":"1.2.0","customMdnsConfig":false,"mdnsMulticast":true,"mdnsInterface":"","mdnsPort":"","mdnsIp":"","mdnsTtl":"","mdnsLoopback":true,"mdnsReuseAddr":true,"allowMessagePassthrough":false}]

**Comment by Nazarbayevv** [comment]
> Could you send an export of your flow ?

[{"id":"276cd89c880e0727","type":"deconz-input","z":"62a84518.fc2e2c","name":"","topic":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","outputs":1,"output_rules":[{"type":"homekit","format":"single","onstart":true,"onerror":true,"payload":["CurrentPosition","TargetPosition"],"uniqueId":"ui-id-11"}],"x":150,"y":480,"wires":[["c5e9ee34a4573cd3","37b4a55ad6577a83"]]},{"id":"0745d4f5296d7bae","type":"deconz-output","z":"62a84518.fc2e2c","name":"","config_version":2,"server":"da76db0b.163948","statustext":"","statustext_type":"default","search_type":"device","device_list":["lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"],"device_name":"","query":"{}","commands":[{"arg":{"payload":{"type":"msg","value":"payload"},"transitiontime":{"type":"num","value":""},"retryonerror":{"type":"num","value":"0"},"aftererror":{"type":"continue","value":""}},"type":"homekit","domain":"","uniqueId":"ui-id-10"}],"specific":{"delay":{"type":"num","value":"50"},"result":{"type":"at_end","value":""}},"x":610,"y":480,"wires":[["908997aa4d3bfc5b"]]},{"id":"c5e9ee34a4573cd3","type":"debug","z":"62a84518.fc2e2c","name":"1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":300,"y":400,"wires":[]},{"id":"59fb1951dfad846a","type":"debug","z":"62a84518.fc2e2c","name":"2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":540,"y":400,"wires":[]},{"id":"908997aa4d3bfc5b","type":"debug","z":"62a84518.fc2e2c","name":"3","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":790,"y":400,"wires":[]},{"id":"37b4a55ad6577a83","type":"homekit-service","z":"62a84518.fc2e2c","isParent":true,"hostType":"0","bridge":"c887541e.9bc998","accessoryId":"","parentService":"","name":"test","serviceName":"WindowCovering","topic":"","filter":false,"manufacturer":"NRCHKB","model":"1.4.3","serialNo":"Default Serial Number","firmwareRev":"1.4.3","hardwareRev":"1.4.3","softwareRev":"1.4.3","cameraConfigVideoProcessor":"ffmpeg","cameraConfigSource":"","cameraConfigStillImageSource":"","cameraConfigMaxStreams":2,"cameraConfigMaxWidth":1280,"cameraConfigMaxHeight":720,"cameraConfigMaxFPS":10,"cameraConfigMaxBitrate":300,"cameraConfigVideoCodec":"libx264","cameraConfigAudioCodec":"libfdk_aac","cameraConfigAudio":false,"cameraConfigPacketSize":1316,"cameraConfigVerticalFlip":false,"cameraConfigHorizontalFlip":false,"cameraConfigMapVideo":"0:0","cameraConfigMapAudio":"0:1","cameraConfigVideoFilter":"scale=1280:720","cameraConfigAdditionalCommandLine":"-tune zerolatency","cameraConfigDebug":false,"cameraConfigSnapshotOutput":"disabled","cameraConfigInterfaceName":"","characteristicProperties":"{}","waitForSetupMsg":false,"outputs":2,"x":370,"y":480,"wires":[["59fb1951dfad846a","0745d4f5296d7bae"],[]]},{"id":"da76db0b.163948","type":"deconz-server","name":"Deconz","config_version":1,"ip":"192.168.88.9","port":"80","ws_port":"443","secure":false,"polling":"15"},{"id":"c887541e.9bc998","type":"homekit-bridge","bridgeName":"Node-RED_Access","pinCode":"400-44-000","port":"","allowInsecureRequest":false,"manufacturer":"NRCHKB","model":"1.2.0","serialNo":"Default Serial Number","firmwareRev":"1.2.0","hardwareRev":"1.2.0","softwareRev":"1.2.0","customMdnsConfig":false,"mdnsMulticast":true,"mdnsInterface":"","mdnsPort":"","mdnsIp":"","mdnsTtl":"","mdnsLoopback":true,"mdnsReuseAddr":true,"allowMessagePassthrough":false}]

**Comment by Nazarbayevv** [comment]
> Could you try this :
> 
> Close the windows cover. Clear all logs Open the cover with the homekit app. Wait 10 seconds after the cover stop moving. Then send all logs you get from the nodes.

![image](https://user-images.githubusercontent.com/50654913/151706943-2c4915dd-04f1-43a0-a8a4-6f642f98e531.png)

1. node: 2
{"TargetPosition":100}

2. node: 3
[{"result":{"lift":100},"request":{"lift":100},"meta":{"etag":"abd929da9e2df64c4a7be025097a81f5","hascolor":false,"lastannounced":"2022-01-07T16:03:30Z","lastseen":"2022-01-07T16:03Z","manufacturername":"_TZE200_xuzcvlku","modelid":"TS0601","name":"Занавеска","state":{"alert":"none","bri":254,"lift":100,"on":true,"open":false,"reachable":true},"swversion":null,"type":"Window covering device","uniqueid":"5c:02:72:ff:fe:e1:60:3f-01","device_type":"lights","device_id":44,"device_path":"lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"}}]

3. node: 1
{"CurrentPosition":1,"TargetPosition":1}


Cover trying to open, but stoping on 1%

**Comment by Nazarbayevv** [comment]
Other example: my winows cover was opened on 8% and i tried to open on 29%. My winows cover closed:
![image](https://user-images.githubusercontent.com/50654913/151707277-d7d10b41-7c0b-401a-ab87-41f270abe4d9.png)

1.
{"TargetPosition":29}
2. [{"result":{"lift":100},"request":{"lift":100},"meta":{"etag":"4329753f1cfe3c38c62f420781cd27e7","hascolor":false,"lastannounced":"2022-01-07T16:03:30Z","lastseen":"2022-01-07T16:03Z","manufacturername":"_TZE200_xuzcvlku","modelid":"TS0601","name":"Занавеска","state":{"alert":"none","bri":233,"lift":92,"on":true,"open":true,"reachable":true},"swversion":null,"type":"Window covering device","uniqueid":"5c:02:72:ff:fe:e1:60:3f-01","device_type":"lights","device_id":44,"device_path":"lights/uniqueid/5c:02:72:ff:fe:e1:60:3f-01"}}]
3. {"CurrentPosition":0,"TargetPosition":0}
4. {"CurrentPosition":7,"TargetPosition":7}
5. {"CurrentPosition":5,"TargetPosition":5}
6. {"CurrentPosition":3,"TargetPosition":3}
7. {"CurrentPosition":1,"TargetPosition":1}
8. {"CurrentPosition":0,"TargetPosition":0}

**Comment by Zehir** [comment]
I don't undestand what is not working properly. Could you change the debug nodes to display the complete msg instead of msg.payload ?

**Comment by Zehir** [comment]
I found the issue, could you test with version 2.3.3 ?

**Comment by Nazarbayevv** [comment]
My issue was fixed in 2.3.3. Thank you.
